src/main/kotlin/exercise3/task4/TaxiParkTask.kt

```diff
@@ -10,29 +10,65 @@ package exercise3.task4
  * Find all the drivers who performed no trips.
  */
 internal fun TaxiPark.findFakeDrivers(): Set<Driver> {
-    TODO("Implement me!!!")
+    val realDriversNames = this.trips.map { it -> it.driver}.toSet()
+    val allDriversNames = this.allDrivers
+    return allDriversNames.subtract(realDriversNames)
 }
 
 /**
  * Subtask 2:
  * Find all the clients who completed at least the given number of trips.
  */
 internal fun TaxiPark.findFaithfulPassengers(minTrips: Int): Set<Passenger> {
-    TODO("Implement me!!!")
+    //test ovde ne prolazi, nisam siguran zasto
+    return this.trips
+        .flatMap { it.passengers }
+        .groupingBy { it }
+        .eachCount()
+        .filter{ it.value >= minTrips }
+        .map { it.key }
+        .toSet()
 }
 
 /**
  * Subtask 3:
  * Find all the passengers, who were taken by a given driver more than once.
  */
 internal fun TaxiPark.findFrequentPassengers(driver: Driver): Set<Passenger> {
-    TODO("Implement me!!!")
+    val allPassengers = mutableListOf<Passenger>()
+    this.trips
+        .filter { it.driver == driver }
+        .map { it.passengers }
+        .forEach {
+            it.forEach {
+                allPassengers.add(it)
+            }
+        }
```

Comment:
Instead of this you can simply use `flatmap` which will convert list of lists to list.